### PR TITLE
[14.0] Add edi_exchange_type_auto

### PR DESCRIPTION
**Pull request by simahawk**
Ease automation of EDI exchange types within the EDI framework.

### CODE REVIEW

**Review by sebalix — COMMENTED** [review]
LG, lot of TODO to address I guess

File: edi_exchange_type_auto/models/edi_auto_exchange_consumer_mixin.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
+# Copyright 2022 Camptocamp SA
```
Comment (TDu) [inline edi_exchange_type_auto/models/edi_auto_exchange_consumer_mixin.py:1]:
```suggestion
# Copyright 2023 Camptocamp SA
```

Comment (simahawk) [reply edi_exchange_type_auto/models/edi_auto_exchange_consumer_mixin.py:1]: it started in 2022 :wink:

**Comment by simahawk** [comment]
spilt core changes to https://github.com/OCA/edi/pull/778
Now I move this PR to OCA/edi-framework.

**Pull request closed without merging**